shared/strbuf.c

```diff
@@ -18,10 +19,20 @@ static bool buf_realloc(struct strbuf *buf, size_t sz)
 {
 	void *tmp;
 
-	tmp = realloc(buf->bytes, sz);
-	if (sz > 0 && tmp == NULL)
-		return false;
+	if (!buf->heap)
+		tmp = malloc(sz);
+	else
+		tmp = realloc(buf->bytes, sz);
```

Comment:
For any fans of the ternary operator, this could be written as `tmp = realloc(!buf->heap ? NULL : buf->bytes, sz);`.

Reply:
I tolerate it, I'm not a fan :).... here I'm ok if we flip them:  `tmp = realloc(buf->heap ? buf->bytes : NULL, sz);`